cheatsheets/Forgot_Password_Cheat_Sheet.md

```diff
@@ -1,60 +1,104 @@
+OTP is the best method in order to implement a secure forgot password service that triggers as a 2FA functionality.
 
-The user should always be sent a token rather than a new password for their account. There are numerous problems with directly sending the user a password, especially via email. These include:
+The two most famous methods are Time-OTP ([TOTP](https://tools.ietf.org/html/rfc6238)), or HMAC-OTP ([HOTP](https://tools.ietf.org/html/rfc4226)). The main difference is in the counter, where TOTP focuses on the Unix time, and HOTP has a counter that gets incremented on every user call to generate the OTP.
 
-- The transport mechanism used to send the password (email, SMS, etc) may not be encrypted.
-- The password may be stored in the users' mailbox or device for an indefinite period.
-- The password may be accessible to IT staff or other users who have access to the mailbox or device.
-- The password may be logged by third parties such as mail security scanning platforms.
+For a better description of OTP generation, refer to the [MFA CS](Multifactor_Authentication_Cheat_Sheet.md#something-you-have), where the pros and cons of every implementation are provided.
 
-Of course, by all means, once a user's password has been reset, the randomly-generated token should no longer be valid.
+One implementation can be found over for [Authy](https://www.twilio.com/docs/authy/tutorials/two-factor-authentication-python-flask). If you don't want to rely on applications (such as Authy, Google/Microsoft Authenticator, etc.), you can generate QR codes and send them from the server for the user to scan using their device. The following python library, [pyotp](https://github.com/pyauth/pyotp), helps the developer implement any of the methods discussed in this section.
 
-## Step 4) Allow user to change password in the existing session
+> OTPs can be sent through other channels as well, such as emails and SMSs. Various attacks and weaknesses have been identified in SMS that it is preferrable not to use them for OTPs.
 
-Step 4 requires input of the code sent in step 3 in the existing session where the challenge questions were answered in step 2, and allows the user to reset his password. Display a simple HTML form with one input field for the code, one for the new password, and one to confirm the new password. Verify the correct code is provided and be sure to enforce all password complexity requirements that exist in other areas of the application. 
+### URL Tokens
```

Comment:
This is essentially a one time password just placed in the URL. An implementation of this could just be an actual OTP added to a query string in a URL. Does it make sense to group these methods together and mention that generation and use can be a little different? Or it might be better keeping these separated, I'm just raising the similarity.